### PR TITLE
Added workaround for MSVC 2013 compiler bug.

### PR DESCRIPTION
**Pull request by neilmacintosh**
This PR should fix the remaining compiler crash reported in #159.

### CODE REVIEW

File: include/string_span.h [hunk 1]
```diff
@@ -22,6 +22,16 @@
 #include "span.h"
 #include <cstring>
 
+// VS 2013 workarounds
+#ifdef _MSC_VER
+#if _MSC_VER <= 1800
+
+#pragma push_macro("GSL_MSVC_HAS_TYPE_DEDUCTION_BUG")
```
Comment (gdr-at-ms) [inline include/string_span.h]: Do you expect many definitions of `GSL_MSVC_HAS_TYPE_DEDUCTION_BUG`?
If not, then the push here will hide conflicts that should be loudly reported to consumers if they ever step on it.

File: include/string_span.h [hunk 2]
```diff
@@ -178,4 +213,15 @@ template <size_t Max = dynamic_range>
 using wzstring_builder = basic_zstring_builder<wchar_t, Max>;
 }
 
+// VS 2013 workarounds
+#ifdef _MSC_VER
+#if _MSC_VER <= 1800
+
+#pragma pop_macro("GSL_MSVC_HAS_TYPE_DEDUCTION_BUG") 
+#undef GSL_MSVC_HAS_TYPE_DEDUCTION_BUG 
+
```
Comment (gdr-at-ms) [inline include/string_span.h]: The `#undef` after the pop will remove any previous binding that the push was attempting to preserve.
I think it is better to just remove the push and the pops.

**Comment by rkhlebnikov** [comment]
Please add `inline` to the definitions - otherwise linker errors are imminent :) Thanks for the workaround!

**Comment by rkhlebnikov** [comment]
Actually, added a pull request for it.

File: include/string_span.h (hunk 2 above)
Comment (neilmacintosh) [inline include/string_span.h]: Good point. I'm going to clean up all the config macros in a separate pass shortly. Given no one else should be using it anyway it's harmless for now.

**Pull request merged**